server/middleware/profile/index.js

```diff
@@ -0,0 +1,158 @@
+const {
+  HttpStatusCodes: { StatusCodes },
+} = require('../../utils/constants');
+const { OK, BAD_REQUEST, GATEWAY_TIMEOUT } = StatusCodes;
+const mongoUtil = require('../../utils/mongoUtil');
+const bcrypt = require('bcryptjs');
+const generateCookie = require('../../utils/generateCookie');
+
+const resObj = (statusCode, message, data = {}) => ({
+  statusCode,
+  message,
+  data,
+});
+
+const createPayload = (body) => {
+  const { email, firstName, lastName, title, hasOnboarded } = body;
+  return {
+    ...(email ? { email: email.toLowerCase() } : {}),
+    ...(firstName ? { firstName: firstName } : {}),
+    ...(lastName ? { lastName: lastName } : {}),
+    ...(title ? { title: title } : {}),
+    ...(hasOnboarded ? { hasOnboarded: hasOnboarded } : {}),
+  };
+};
+
+const updateProfileData = async (req, res, tokenData) => {
+  let data;
+  let statusCode;
+  let message;
+  try {
+    const userPayload = createPayload(req.body);
+    const { USERS_COLLECTION } = process.env;
+    const userCollection = await mongoUtil.fetchCollection(USERS_COLLECTION);
+    const updateProfile = await userCollection.findOneAndUpdate(
+      { email: tokenData.email },
+      { $set: userPayload },
+      { returnOriginal: false }
+    );
+    if (!updateProfile.value) {
+      statusCode = 401;
+      throw Error('User does not exist!');
+    }
+    const { password, ...rest } = updateProfile.value;
+    generateCookie(res, rest);
+    statusCode = OK;
+    message = 'Success';
+    data = rest;
+  } catch (err) {
+    if (err) {
+      statusCode = statusCode || BAD_REQUEST;
+      message = err.message;
+    } else {
+      statusCode = GATEWAY_TIMEOUT;
+      message = 'Gateway timeout';
+    }
```

Comment:
log error events

Reply:
resolve in commit #https://github.com/femmecubator/femmecubator-api-repo/pull/72/commits/58b6dd83fe0bda2e2c2dc2899c8593f45bc7bcc0